cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_slurm_config_generator.py

```diff
@@ -148,7 +148,8 @@ def __init__(self, queue_name: str, compute_resource_config: Dict, no_gpu, memor
         self.static_nodes = compute_resource_config["MinCount"]
         self.dynamic_nodes = compute_resource_config["MaxCount"] - self.static_nodes
         self.name = compute_resource_config["Name"]
-        self.disable_simultaneous_multithreading = compute_resource_config["DisableSimultaneousMultithreading"]
+        self.disable_multithreading = compute_resource_config.get("DisableSimultaneousMultithreading", False)
```

Comment:
> we kept the CustomSlurmSettings at compute resource and queue level with a default value of {}
Since It was already in the code, replacing it with `None` wouldn't change the logic or add anything else, so I kept it as it was.

And also here: this `.get("CustomSlurmSettings", {})` this ensure we fall back to the same default.